### PR TITLE
initial checkin

### PR DESCRIPTION
**Pull request by bwalsh**
This PR contains the following features:
* a Dockerfile to run the harvesters
* instructions to build, run, and copy the results to swift
* by default the harvesters will create /output/[harvester].json with one line per json object

### CODE REVIEW

**Comment by prismofeverything** [comment]
This looks great, straightforward and simple. I see most of it is there to call code that exists elsewhere. Could this be the basis for a kind of template for extractors/transformers like this?

As in: all we need to know to run one of these is - 
* Dockerfile creating a container (and presumably setting up the extract)
* what input files are required, and where do they come from?
* how do you call the container with the input files?
* what outputs are created?

(Kyle I know what you are going to say: that sounds like a workflow. But I don't think we need a full workflow to do this, something simpler and more specific could suffice and make it actually happen).

Once these exist, we could have a registry that tracks them and is able to run them all (even with a dependency graph so that you could expire one and have it refetch its source and recompute, potentially triggering downstream expiration etc). 

Gah, I can see it in my mind.

Thoughts or ideas on this? I might throw together a prototype soon.

**Comment by bwalsh** [comment]
I've added a cwl description yml file.

Please note:  The volumes clause in the yml file is implementation dependent.

File: g2p-transform.cwl [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+outputs:
+  output:
+    type: Directory
+volumes:
+ - name: output
+   hostPath: /tmp/g2p
+   swiftPath: $OS_STORAGE_URL/etl-development/g2p
+ - name: /root/.synapseCache/.session
+   hostPath: ~/.synapseCache/.session
```
Comment (kellrott) [inline g2p-transform.cwl:42]: I think you are thinking about 'input binding' ( http://www.commonwl.org/v1.0/CommandLineTool.html#CommandLineBinding )
CWL specifies that the engine decides how to mount files into container path

File: g2p-transform.cwl [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    type: File
+
+outputs:
+  output:
+    type: Directory
+volumes:
+ - name: output
+   hostPath: /tmp/g2p
+   swiftPath: $OS_STORAGE_URL/etl-development/g2p
+ - name: /root/.synapseCache/.session
```
Comment (kellrott) [inline g2p-transform.cwl:41]: It seems weird to pull in the Synapse client session info to the docker container

File: g2p-transform.cwl [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+# -e "MOLECULAR_MATCH_API_KEY=XXXXXXXX" \
+# -v /tmp/g2p:/output \
+# -v ~/.synapseCache/.session:/root/.synapseCache/.session  \
+# -v /tmp/g2p/cgi_biomarkers_per_variant.tsv:/g2p-aggregator/harvester/cgi_biomarkers_per_variant.tsv \
+# -v /tmp/g2p/cosmic_lookup_table.tsv:/g2p-aggregator/harvester/cosmic_lookup_table.tsv  \
+# -v /tmp/g2p/harvester.sqlite:/g2p-aggregator/harvester/harvester.sqlite \
+# -v /tmp/g2p/oncokb_all_actionable_variants_20170822.tsv:/g2p-aggregator/harvester/oncokb_all_actionable_variants.tsv \
+# g2p-transform
+
+# note: the volumes & swiftPath attribute is somewhat ad-hoc, there is no official support for openstack swift
```
Comment (kellrott) [inline g2p-transform.cwl:14]: I think at this level, it should be more of an object name, the fact that swift or s3 is behind the scenes shouldn't matter

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+COPY ./harvest-all.sh /g2p-aggregator/harvester
+ENTRYPOINT ["./harvest-all.sh"]
+
+
+# documentation
+LABEL maintainer="walsbr@ohsu.edu"
+LABEL org.label-schema.name="g2p"
+LABEL org.label-schema.description="Harvest genotype and phenotype data."
+LABEL org.label-schema.usage="https://github.com/biostream/g2p-transform/blob/master/README.md"
+LABEL org.label-schema.docker.cmd="docker run --rm -it \
```
Comment (kellrott) [inline Dockerfile:27]: Is this for documentation, or something that is going to be used?

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+RUN git clone https://github.com/ohsu-comp-bio/g2p-aggregator.git
+WORKDIR /g2p-aggregator
+RUN git checkout new-harvesters
+
+# install dependencies
+WORKDIR /g2p-aggregator/harvester
+RUN pip install -r requirements.txt
+
+# output should appear here
+VOLUME ["/output"]
```
Comment (kellrott) [inline Dockerfile:15]: CWL defines where the outputs are supposed to be

File: g2p-transform.cwl (hunk 2 above)
Comment (bwalsh) [reply g2p-transform.cwl:41]: Fair point:  I can experiment & change it to `session_token`
http://docs.synapse.org/python/Client.html#synapseclient.Synapse.login

**Pull request merged**